packages/dev/core/src/Engines/thinEngine.ts

```diff
@@ -2935,8 +2936,26 @@ export class ThinEngine {
         indexParameters?: any,
         shaderLanguage = ShaderLanguage.GLSL
     ): Effect {
-        const vertex = baseName.vertexElement || baseName.vertex || baseName.vertexToken || baseName.vertexSource || baseName;
-        const fragment = baseName.fragmentElement || baseName.fragment || baseName.fragmentToken || baseName.fragmentSource || baseName;
+        const vertex =
+            typeof baseName === "string"
+                ? baseName
+                : "vertexElement" in baseName
+                ? baseName.vertexElement
+                : "vertex" in baseName
+                ? baseName.vertex
+                : "vertexToken" in baseName
+                ? baseName.vertexToken
+                : baseName.vertexSource || baseName;
+        const fragment =
+            typeof baseName === "string"
+                ? baseName
+                : "fragmentElement" in baseName
+                ? baseName.fragmentElement
+                : "fragment" in baseName
+                ? baseName.fragment
+                : "fragmentToken" in baseName
+                ? baseName.fragmentToken
+                : baseName.fragmentSource || baseName;

```

Comment:
If someone knows of a shorter way of making Typescript happy, we could do that.

Also, this technically changes the behaviour. Previously `{ vertexElement: "", vertex: "custom" }` would have caused the code to read "custom". Now it reads `""`.